### PR TITLE
DOC: Modify IntervalArray/IntervalIndex docstrings to pass validate_docstrings.py

### PR DESCRIPTION
**Pull request by jschendel**
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Didn't materially change any existing content; mostly small modifications and a couple additions.

### CODE REVIEW

File: pandas/core/arrays/interval.py [hunk 1]
```diff
@@ -87,18 +90,34 @@
...
-    name='', extra_methods='', examples='',
+    name='',
+    extra_methods='',
+    examples=textwrap.dedent("""\
+    Examples
+    --------
+    A new ``IntervalArray`` can be constructed directly from an array-like of
+    ``Interval`` objects:
+
+    >>> pd.core.arrays.IntervalArray([pd.Interval(0, 1), pd.Interval(1, 5)])
```
Comment (jorisvandenbossche) [inline pandas/core/arrays/interval.py]: I think we should keep this out of the docs until we know if or how to expose this publicly (which will in any case not be `pd.core.arrays.IntervalArray`)

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good! 

Small comment about adding an actual `IntervalArray` example to the docstrings.

Do the doctests now pass? 
In that case, can you add this file to the doctests that are run in `ci/code_checks.sh` ?

File: pandas/core/arrays/interval.py (hunk 1 above)
Comment (jschendel) [reply pandas/core/arrays/interval.py]: `ci/code_checks.sh` won't check the `IntervalArray` docstring if the Examples section is missing.  I've added a TODO note to myself to modify the example once `IntervalArray` has been made public; can open an issue for it as well if need be.

The issue for making extension arrays public has a milestone of 0.24.0, so I should be able to address this prior to release, and it should just be a matter of pasting the updated call. Can remove the example though, if you'd still prefer that approach.

**Comment by jschendel** [comment]
Added `indexes/interval.py` and `arrays/interval.py` to the ``ci/code_checks.sh`.  Had to make it ignore some methods that aren't passing yet, but will get to those shortly in a separate PR.

File: pandas/core/arrays/interval.py (hunk 1 above)
Comment (jschendel) [reply pandas/core/arrays/interval.py]: ping @jorisvandenbossche : thoughts on the above?

Comment (jorisvandenbossche) [reply pandas/core/arrays/interval.py]: OK, let's decide about it / clean-up later

**Pull request merged**